modules/datadog-monitors.nix

```diff
@@ -0,0 +1,69 @@
+with (import ./../lib.nix);
+
+rec {
+  mkMonitor = { name, type, query, message ? "", monitorOptions ? {} }:
```

Comment:
Looks good :) I'd like to avoid this pattern of functions that return nixos modules, but rather use options in modules to enable monitors.

Something like `iohk.monitors = [ disk_monitor ntp_monitor ];`. @cleverca22 can help you with options, but we are using them for example in `modules/cardano-node.nix`